CiE-iOS/CiE-iOS/ViewControllers/LoginViewController.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  LoginViewController.swift
+//  CiE-iOS
+//
+//  Created by Jack Tan on 20/5/18.
+//  Copyright © 2018 Nikita Hans. All rights reserved.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
+    
+    @IBOutlet weak var usernameField: UITextField!
+    @IBOutlet weak var passwordField: UITextField!
+    
+    @IBAction func LoginButtonPressed(_ sender: Any) {
+        //print("Button pressed!")
+        self.performSegue(withIdentifier: "Tab Bar Controller", sender: self)
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
```

Comment:
Not sure but if I red correctly you are required to override all 4 versions of this function not just one.
I am not sure but please make sure you know what you are doing.
In my experience to trigger something at every touchesBegan can make handling other behaviour when touches Begin difficult. Try to find a way in that the UITextField is bound the required behaviour not the entire View.